dali/python/nvidia/dali/pipeline.py

```diff
@@ -1364,3 +1369,39 @@ def create_pipeline(*args, **kwargs):
             return pipe
         return create_pipeline
     return actual_decorator(fn) if fn else actual_decorator
+
+
+def _pipeline_def_experimental(fn=None, **pipeline_kwargs):
+    from nvidia.dali._debug_mode import _PipelineDebug
+    pipeline_debug =  pipeline_kwargs.pop('debug', False)
+    def actual_decorator(func):
+        @functools.wraps(func)
+        def create_pipeline(*args, **kwargs):
+            debug_mode_on = kwargs.get('debug', pipeline_debug)
+            ctor_args, fn_kwargs = _discriminate_args(func, **kwargs)
+            if debug_mode_on:
+                pipe = _PipelineDebug(functools.partial(func, *args, **fn_kwargs),
+                                     **{**pipeline_kwargs, **ctor_args})
+            else:
+                pipe = Pipeline(**{**pipeline_kwargs, **ctor_args})  # Merge and overwrite dict
```

Comment:
It's your call, but this part `**{**pipeline_kwargs, **ctor_args}` is a magic (that's why I've put comment there). I feel that extracting it might make the code more readable:
```python
            ctor_args, fn_kwargs = _discriminate_args(func, **kwargs)
            pipeline_args = **{**pipeline_kwargs, **ctor_args}  # Merge and overwrite dict
            if debug_mode_on:
                pipe = _PipelineDebug(functools.partial(func, *args, **fn_kwargs),
                                     pipeline_args)
            else:
                pipe = Pipeline(pipeline_args)
```

Reply:
done